systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectST.java

```diff
@@ -992,6 +1002,80 @@ void testScaleConnectAndConnectorSubresource() {
+            .build();
+
+        ConfigMap configMap = new ConfigMapBuilder()
+            .editOrNewMetadata()
+                .withName("my-config-map")
+            .endMetadata()
+            .addToData("my-key", "my-value")
+            .build();
+
+        kubeClient().createSecret(connectSecret);
+        kubeClient().getClient().configMaps().inNamespace(NAMESPACE).createOrReplace(configMap);
+
+        KafkaResource.kafkaEphemeral(CLUSTER_NAME, 3).done();
+
+        KafkaConnectResource.kafkaConnect(CLUSTER_NAME, 1)
+            .editMetadata()
+                .addToAnnotations(Annotations.STRIMZI_IO_USE_CONNECTOR_RESOURCES, "true")
+            .endMetadata()
+            .editSpec()
+                .withNewExternalConfiguration()
+                    .addNewVolume()
+                        .withNewName("connect-secret")
+                        .withSecret(new SecretVolumeSourceBuilder().withSecretName("my-secret").build())
+                    .endVolume()
+                    .addNewVolume()
+                        .withNewName("connect-config-map")
+                        .withConfigMap(new ConfigMapVolumeSourceBuilder().withName("my-config-map").build())
+                    .endVolume()
+                    .addNewEnv()
+                        .withNewName(secretEnv)
+                        .withNewValueFrom()
+                            .withSecretKeyRef(
+                                new SecretKeySelectorBuilder()
+                                    .withKey("my-secret-key")
+                                    .withName(connectSecret.getMetadata().getName())
+                                    .withOptional(false)
+                                    .build())
+                        .endValueFrom()
+                    .endEnv()
+                    .addNewEnv()
+                        .withNewName(configMapEnv)
+                        .withNewValueFrom()
+                            .withConfigMapKeyRef(
+                                new ConfigMapKeySelectorBuilder()
+                                    .withKey("my-key")
+                                    .withName(configMap.getMetadata().getName())
+                                    .withOptional(false)
+                                    .build())
+                        .endValueFrom()
+                    .endEnv()
+                .endExternalConfiguration()
+            .endSpec()
+            .done();
+
+        String connectPodName = kubeClient().listPods(Labels.STRIMZI_KIND_LABEL, KafkaConnect.RESOURCE_KIND).get(0).getMetadata().getName();
+
+        LOGGER.info("Check if the ENVs are correctly contains desired values");
+        assertThat(cmdKubeClient().execInPod(connectPodName, "/bin/bash", "-c", "printenv " + secretEnv).out().trim(), equalTo(secretPassword));
+        assertThat(cmdKubeClient().execInPod(connectPodName, "/bin/bash", "-c", "printenv " + configMapEnv).out().trim(), equalTo("my-value"));
```

Comment:
You check just the env vars. Not the volumes. You should also check the volumes (e.g. `cat ....`)

Reply:
Sure thing, thanks :)

Reply:
Done